skyportal/tests/api/test_photometry.py

```diff
@@ -92,35 +158,162 @@ def test_token_user_update_photometry(upload_data_token,
     photometry_id = data['data']['ids'][0]
     status, data = api(
         'GET',
-        f'photometry/{photometry_id}',
+        f'photometry/{photometry_id}?format=flux&magsys=vega',
         token=upload_data_token)
     assert status == 200
     assert data['status'] == 'success'
-    assert data['data']['flux'] == 12.24
+
+    ab = sncosmo.get_magsystem('ab')
+    vega = sncosmo.get_magsystem('vega')
+    correction = 2.5 * np.log10(vega.zpbandflux('ztfg') / ab.zpbandflux('ztfg'))
+
+
+    np.testing.assert_allclose(data['data']['flux'], 10**(-0.4 * (21 - correction - 23.9)))
+    np.testing.assert_allclose(data['data']['fluxerr'], 0.2 / (2.5 / np.log(10)) * data['data']['flux'])
+    np.testing.assert_allclose(data['data']['zp'], 23.9 + correction)
 
     status, data = api(
-        'PUT',
-        f'photometry/{photometry_id}',
-        data={'flux': 11.0},
-        token=manage_sources_token)
+        'GET',
+        f'photometry/{photometry_id}?format=flux&magsys=ab',
+        token=upload_data_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    np.testing.assert_allclose(data['data']['flux'], 10**(-0.4 * (21 - correction - 23.9)))
+    np.testing.assert_allclose(data['data']['fluxerr'], 0.2 / (2.5 / np.log(10)) * data['data']['flux'])
+    np.testing.assert_allclose(data['data']['zp'], 23.9)
+
+
+def test_token_user_mixed_photometry_post(upload_data_token, public_source,
+                                          ztf_camera):
+
+    status, data = api('POST', 'photometry',
+                       data={'obj_id': str(public_source.id),
+                             'mjd': 58000.,
+                             'instrument_id': ztf_camera.id,
+                             'mag': 21.,
```

Comment:
What happens here in this case with a scalar mag value and list of errors?

Reply:
The mag scalar is broadcast  to match the list of errors. Think pandas
broadcasting logic.

Reply:
The API docs describe the broadcasting rules in detail